src/libraries/System.Globalization/tests/CompareInfo/CompareInfoTests.LastIndexOf.cs

```diff
@@ -92,7 +92,13 @@ public static IEnumerable<object[]> LastIndexOf_TestData()
             // ICU matches weightless characters at 1 index prior to the end of the string
             yield return new object[] { s_invariantCompare, "", "\u200d", 0, 0, CompareOptions.None, 0, 0 };
             yield return new object[] { s_invariantCompare, "", "\u200d", -1, 0, CompareOptions.None, 0, 0 };
-            yield return new object[] { s_invariantCompare, "hello", "\u200d", 4, 5, CompareOptions.IgnoreCase, useNls ? 5 : 4 , 0};
+            yield return new object[] { s_invariantCompare, "hello", "\u200d", 4, 5, CompareOptions.IgnoreCase, 5, 0};
+            yield return new object[] { s_invariantCompare, "hello", "\0", 4, 5, CompareOptions.None, useNls ? -1 : 5, 0};
+
+            yield return new object[] { s_invariantCompare, "A\u0303", "\u200d", 1, 2, CompareOptions.None, 2, 0}; // A +  ̃ = Ã
+            yield return new object[] { s_invariantCompare, "A\u0303\u200D", "\u200d", 2, 3, CompareOptions.None, 3, 0}; // A +  ̃ = Ã
+            yield return new object[] { s_invariantCompare, "😁", "\u200d", 1, 2, CompareOptions.None, 2, 0}; // 😁 is the surrogate \u0001F601
```

Comment:
Can we actually use the \u syntax?  We've tried to avoid having non-ASCII in our .cs files.